Sources/PackageSigning/SignatureProvider.swift

```diff
@@ -78,201 +82,189 @@ public enum SignatureStatus: Equatable {
-        var signerStatus = CMSSignerStatus.needsDetachedContent
-        var certificateVerifyResult: OSStatus = errSecSuccess
-        var trust: SecTrust?
-
-        // TODO: build policy based on user config
-        let basicPolicy = SecPolicyCreateBasicX509()
-        let revocationPolicy = SecPolicyCreateRevocation(kSecRevocationOCSPMethod)
-        CMSDecoderCopySignerStatus(
-            cmsDecoder,
-            0,
-            [basicPolicy, revocationPolicy] as CFArray,
-            true,
-            &signerStatus,
-            &trust,
-            &certificateVerifyResult
-        )
+            let signature = try privateKey.sign(content: content, algorithm: self.signatureAlgorithm)
 
-        guard certificateVerifyResult == errSecSuccess else {
-            return .certificateInvalid("Certificate verify result: \(certificateVerifyResult)")
+            do {
+                return try CMS.sign(
+                    signatureBytes: ASN1OctetString(contentBytes: ArraySlice(signature)),
+                    signatureAlgorithm: self.signatureAlgorithm.certificateSignatureAlgorithm,
+                    certificate: try Certificate(secIdentity: secIdentity)
+                )
+            } catch {
+                throw SigningError.signingFailed("\(error)")
+            }
         }
-        guard signerStatus == .valid else {
-            return .invalid("Signer status: \(signerStatus)")
+        #endif
+
+        guard let swiftSigningIdentity = identity as? SwiftSigningIdentity else {
+            throw SigningError.signingIdentityNotSupported
         }
 
-        guard let trust = trust else {
-            return .certificateNotTrusted
+        do {
+            return try CMS.sign(
+                content,
+                signatureAlgorithm: self.signatureAlgorithm.certificateSignatureAlgorithm,
+                certificate: swiftSigningIdentity.certificate,
+                privateKey: swiftSigningIdentity.privateKey
+            )
+        } catch let error as CertificateError where error.code == .unsupportedSignatureAlgorithm {
+            throw SigningError.keyDoesNotSupportSignatureAlgorithm
+        } catch {
+            throw SigningError.signingFailed("\(error)")
         }
+    }
 
-        // TODO: user configured trusted roots
-        SecTrustSetNetworkFetchAllowed(trust, true)
-        //        SecTrustSetAnchorCertificates(trust, trustedCAs as CFArray)
-        //        SecTrustSetAnchorCertificatesOnly(trust, true)
+    func status(
+        signature: [UInt8],
+        content: [UInt8],
+        verifierConfiguration: VerifierConfiguration,
+        observabilityScope: ObservabilityScope
+    ) async throws -> SignatureStatus {
+        let result = await CMS.isValidSignature(
+            dataBytes: content,
+            signatureBytes: signature,
+            trustRoots: CertificateStore(
+                try verifierConfiguration.trustedRoots
+                    .map { try Certificate(derEncoded: $0) }
```

Comment:
parsing the trusted roots can be an expensive operation. Should we do some caching if we need to validate multiple signatures with the same verifier configuration? cc: @Lukasa

Reply:
added caching